server/src/test/java/org/apache/druid/metadata/SQLMetadataConnectorTest.java

```diff
@@ -305,6 +307,170 @@ public void testIsTransientException()
     );
   }
 
+  @ParameterizedTest
+  @ValueSource(booleans = {true, false})
+  public void test_tableIndices_areNotReplaced_ifExist(boolean useUniqueIndexName)
+  {
+    tablesConfig = new MetadataStorageTablesConfig(
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        useUniqueIndexName
+    );
+    connector = new TestDerbyConnector(new MetadataStorageConnectorConfig(), tablesConfig);
+
+    final String segmentsTable = tablesConfig.getSegmentsTable();
+    final List<String> legacyIndexTemplates = Arrays.asList(
+        "idx_%1$s_used",
+        "idx_%1$s_datasource_used_end_start",
+        "idx_%1$s_datasource_upgraded_from_segment_id"
+    );
+    final List<List<String>> indexColumns = List.of(
+        List.of("used"),
+        List.of(
+            "dataSource",
+            "used",
+            StringUtils.format("%send%S", connector.getQuoteString(), connector.getQuoteString()),
+            "start"
+        ),
+        List.of("dataSource", "upgraded_from_segment_id")
+    );
+    final List<String> legacyIndexNames = legacyIndexTemplates.stream()
+                                                              .map(template -> StringUtils.toUpperCase(StringUtils.format(
+                                                                  template,
+                                                                  segmentsTable
+                                                              )))
+                                                              .collect(Collectors.toList());
+    final List<String> uniqueIndexNames = indexColumns.stream()
+                                                      .map(cols -> connector.generateUniqueIndexName(
+                                                          segmentsTable,
+                                                          cols
+                                                      ))
+                                                      .collect(Collectors.toList());
+
+    connector.createSegmentTable(segmentsTable);
+    connector.alterSegmentTable();
+
+    // drop legacy/unique indices and create the other type
+    connector.getDBI().withHandle(handle -> {
+      for (int i = 0; i < legacyIndexNames.size(); i++) {
+        handle.execute(StringUtils.toUpperCase(StringUtils.format(
+            "DROP INDEX %s",
```

Comment:
We should avoid these formatters and instead just drop some of the indexes
(preferably not all so that we can verify the difference in behaviour)

e.g. `DROP INDEX IDX_ABC`.